synapseclient/models/mixins/table_components.py

```diff
@@ -442,6 +452,147 @@ async def store_async(
+            `Table(columns=[Column(name="foo")])`, they will be added at the begining
+            of the columns list.
+        - If you use the `store_rows` method and the `schema_storage_strategy` is set to
+            `INFER_FROM_DATA` the columns will be added at the end of the columns list.
+
+
+        Arguments:
+            dry_run: If True, will not actually store the table but will log to
+                the console what would have been stored.
+            job_timeout: The maximum amount of time to wait for a job to complete.
+                This is used when updating the table schema. If the timeout
+                is reached a `SynapseTimeoutError` will be raised.
+                The default is 600 seconds
+            synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+
+
+        Returns:
+            The View instance stored in synapse.
+        """
+        client = Synapse.get_client(synapse_client=synapse_client)
+
+        if self.include_default_columns:
+            view_type_mask = None
+            if self.view_type_mask:
+                if isinstance(self.view_type_mask, ViewTypeMask):
+                    view_type_mask = self.view_type_mask.value
+                else:
+                    view_type_mask = self.view_type_mask
+
+            default_columns = await get_default_columns(
+                view_entity_type=(
+                    self.view_entity_type if self.view_entity_type else None
+                ),
+                view_type_mask=view_type_mask,
+                synapse_client=synapse_client,
+            )
+            for default_column in default_columns:
+                if (
+                    default_column.name in self.columns
+                    and default_column != self.columns[default_column.name]
+                ):
+                    client.logger.warning(
+                        f"Column '{default_column.name}' already exists in dataset. "
+                        "Overwriting with default column."
+                    )
+                self.columns[default_column.name] = default_column
+        # check that column names match this regex "^[a-zA-Z0-9,_.]+"
+        for _, column in self.columns.items():
+            if not re.match(r"^[a-zA-Z0-9,_.]+$", column.name):
+                raise ValueError(
+                    f"Column name '{column.name}' contains invalid characters. "
+                    "Names may only contain: letters, numbers, spaces, underscores, "
+                    "hyphens, periods, plus signs, apostrophes, and parentheses."
+                )
+        return await super().store_async(
+            dry_run=dry_run, job_timeout=job_timeout, synapse_client=synapse_client
+        )
+
+
+@async_to_sync
+class DeleteMixin:
```

Comment:
I imagine these sort of mixins will be relevant across all entity types. Is there something special about Table deletions?

Reply:
I don't think there is anything too specific about Tables/Views in terms of how you delete them. In the future if we want this could be extended to other entity types.